src/Transposit.ts

```diff
@@ -137,40 +135,85 @@ export class Transposit {
     );
   }
 
+  get stash() {
+    return new Stash(this);
+  }
+
   async loadUser(): Promise<User> {
     this.assertIsSignedIn();
 
-    return await trfetch<User>(this.uri("/api/v1/user"), {
-      method: "GET",
-      headers: {
-        "Content-Type": "application/json",
-        Authorization: `Bearer ${this.accessToken}`,
-      },
-    });
+    return await this.makeCallJson<User>("GET", "/api/v1/user");
   }
 
   async run(
     operationId: string,
     parameters: OperationParameters = {},
   ): Promise<EndRequestLog> {
-    const headers: HeadersInit = {
-      "Content-Type": "application/json",
-    };
-    if (this.accessToken) {
-      headers.Authorization = `Bearer ${this.accessToken}`;
-    }
+    return this.makeCallJson<EndRequestLog>(
+      "POST",
+      `/api/v1/execute/${operationId}`,
+      { body: { parameters } },
+    );
+  }
 
-    return await trfetch<EndRequestLog>(
-      this.uri(`/api/v1/execute/${operationId}`),
-      {
-        method: "POST",
-        headers,
-        body: JSON.stringify({
-          parameters,
-        }),
+  async makeCallJson<T>(
+    method: string,
+    path: string,
+    httpParams: HttpParams = {},
+  ): Promise<T> {
+    const response = await this.makeCall(method, path, httpParams);
+    return response.json().then(
+      x => x,
+      () => {
+        throw new APIError("Failed to read response body", response);
       },
     );
   }
+
+  async makeCall(
+    method: string,
+    path: string,
+    { queryParams, body, headers }: HttpParams = {},
+  ): Promise<Response> {
+    if (headers == null) {
```

Comment:
nit: it's typically more common in javascript to do `if (!headers)` instead of using double equals

Reply:
I think I did the explicit null check as a holdover from python where an empty object gets treated as falsy. Fortunately, javascript doesn't seem to do this.

Also, someone could have passed in headers = false, though with the typing change I don't need to care about that.